component/src/main/java/org/wso2/extension/siddhi/map/json/sinkmapper/JsonSinkMapper.java

```diff
@@ -130,36 +131,49 @@
 
     @Override
     public Class[] getOutputEventClasses() {
-        return new Class[] {String.class};
+        return new Class[]{String.class};
     }
 
     /**
      * Initialize the mapper and the mapping configurations.
      *
-     * @param streamDefinition       The stream definition
-     * @param optionHolder           Option holder containing static and dynamic options
-     * @param templateBuilder Unmapped payload for reference
+     * @param streamDefinition          The stream definition
+     * @param optionHolder              Option holder containing static and dynamic options
+     * @param payloadTemplateBuilderMap Unmapped list of payloads for reference
      */
     @Override
-    public void init(StreamDefinition streamDefinition, OptionHolder optionHolder, TemplateBuilder templateBuilder,
-                     ConfigReader configReader, SiddhiAppContext siddhiAppContext) {
+    public void init(StreamDefinition streamDefinition, OptionHolder optionHolder,
+                     Map<String, TemplateBuilder> payloadTemplateBuilderMap, ConfigReader mapperConfigReader,
+                     SiddhiAppContext siddhiAppContext) {
 
         attributeNameArray = streamDefinition.getAttributeNameArray();
         enclosingElement = optionHolder.validateAndGetStaticValue(ENCLOSING_ELEMENT_IDENTIFIER, null);
         isJsonValidationEnabled = Boolean.parseBoolean(optionHolder
                 .validateAndGetStaticValue(JSON_VALIDATION_IDENTIFIER, "false"));
+
+        //if @payload() is added there must be at least 1 element in it, otherwise a SiddhiParserException raised
+        if (payloadTemplateBuilderMap != null && payloadTemplateBuilderMap.size() != 1) {
+            throw new SiddhiAppCreationException("Json sink-mapper does not support multiple @payload mappings, " +
+                    "error at the mapper of '" + streamDefinition.getId() + "'");
+        }
+        if (payloadTemplateBuilderMap != null &&
+                payloadTemplateBuilderMap.get(payloadTemplateBuilderMap.keySet().iterator().next()).isObjectMessage()) {
+            throw new SiddhiAppCreationException("Json sink-mapper does not support object @payload mappings, " +
+                    "error at the mapper of '" + streamDefinition.getId() + "'");
+        }
     }
 
     @Override
-    public void mapAndSend(Event[] events, OptionHolder optionHolder, TemplateBuilder templateBuilder,
-                           SinkListener sinkListener) {
+    public void mapAndSend(Event[] events, OptionHolder optionHolder,
+                           Map<String,TemplateBuilder> payloadTemplateBuilderMap, SinkListener sinkListener) {
```

Comment:
Map<String,TemplateBuilder> : add a space after the comma. 
This will be failed by checkstyles.

Reply:
There is a new line. So build would not fail. isn't it?

Reply:
I was referring to the comma between String and TemplateBuilder. Locally it failed the build.